rql_download/fuse/create_home_deamon.py

```diff
@@ -8,75 +8,100 @@
 ##########################################################################
 
 """
-> ldpa exemple:
-    python create_home_deamon.py -g OPEN_FU1 -l -b /chroot_fuse -n imagen2 -p mypwd -i 503
+> ldap exemple:
+    create_home_deamon.py -g OPEN_FU1 -l -b /chroot_fuse -n imagen2 -p mypwd -i 503
 """
 # System import
+import pwd
 import grp
 from optparse import OptionParser
 import os
-
-# LDAP import
 import ldap
 
 
-# Parse the command line
-parser = OptionParser()
-parser.add_option("-g", "--gname", dest="group_name",
-                  help="the group name from which we want to retrieve the "
-                       "users (system or ldap).")
-parser.add_option("-l", "--ldap",
-                  action="store_true", dest="use_ldap", default=False,
-                  help="search on ldap.")
-parser.add_option("-b", "--basedir", dest="basedir",
-                  help="the basedir where the user home will be built.")
-parser.add_option("-n", "--dbname", dest="db_name",
-                  help="the cw database instance name.")
-parser.add_option("-a", "--ldapaddr", dest="uri", default="ldap://127.0.0.1",
-                  help="the ldap address.")
-parser.add_option("-c", "--ldapuser", dest="user", default="admin",
-                  help="the ldap user login.")
-parser.add_option("-r", "--ldapbase", dest="base",
-                  default="dc=imagen2,dc=cea,dc=fr",
-                  help="the ldap base login.")
-parser.add_option("-p", "--ldappwd", dest="password",
-                  help="the ldap user password.")
-parser.add_option("-i", "--cwuser", dest="cw_uid",
-                  help="the cw instance user id.")
-(options, args) = parser.parse_args()
+def etc_members(group):
+    try:
+        gr = grp.getgrnam(group)
+    except KeyError:
+        return set()
+    else:
+        return set(gr.gr_mem +
+                   [pw.pw_name for pw in pwd.getpwall()
+                    if pw.pw_gid == gr.gr_gid])
+
+
+def ldap_members(group, ldapobject, group_base, people_base):
+    members = set()
+    filterstr = "(&(objectClass=posixGroup)(cn={0}))".format(group)
+    scope = ldap.SCOPE_ONELEVEL
+    for dn, entry in ldapobject.search_s(group_base, scope, filterstr):
+        group_gid = entry["gidNumber"][0]  # numerical gid
+        if "memberUid" in entry:
+            for member_uid in entry["memberUid"]:
+                members.add(member_uid)
+        filterstr = ("(&(objectClass=inetOrgPerson)"
+                     "(gidNumber={0}))").format(group_gid)
+        for dn, entry in ldapobject.search_s(people_base, scope, filterstr):
```

Comment:
Primary group is found in 'People'.
Secondary groups are found in 'Groups'.
The whole idea was to take into account the primary group.